### DISCUSSION TITLE
Limitting CPU usage

### DISCUSSION THREAD

**Discussion by themisir** [argoproj/argo-cd#5406] [Q&A] [answered]
I've installed argocd using helm chart. I want to limit argocd CPU usage to make cluster stable otherwise argocd sometimes uses too much CPU that causes server overloading, or other issues.

Which configurations do I have to change to limit CPU usage?

**Comment by jannfis** [comment, marked as answer]
Hey @TheMisir, you'd probably want to set resource limits on `argocd-application-controller` workload, which I'd assume is the main consumer of CPU cycles in Argo CD.

This is a Kubernetes setting. You can read more about how to configure CPU limits here: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/

**Reply by jannfis** [reply]
For the Helm chart, the appropriate settings can be made in `controller.resources` value: https://github.com/argoproj/argo-helm/blob/ba467c8a4bc2f0c8a56bb016142cbd90a9e2562b/charts/argo-cd/values.yaml#L119

**Reply by themisir** [reply]
thanks
